apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ts

```diff
@@ -222,56 +213,69 @@ router.get(
       config.secretKey,
     );
 
-    res.locals.questionTestPath = questionTestPath;
-    res.locals.questionTestCsrfToken = questionTestCsrfToken;
-
-    res.locals.isEnterprise = isEnterprise();
+    const isEnterpriseBoolean = isEnterprise();
 
-    res.locals.questionGHLink = null;
+    let questionGHLink: string | null = null;
     if (res.locals.course.repository) {
       const githubRepoMatch = res.locals.course.repository.match(
         /^git@github.com:\/?(.+?)(\.git)?\/?$/,
       );
       if (githubRepoMatch) {
-        res.locals.questionGHLink =
+        questionGHLink =
           'https://github.com/' +
           githubRepoMatch[1] +
           `/tree/${res.locals.course.branch}/questions/` +
           res.locals.question.qid;
       }
     } else if (res.locals.course.example_course) {
-      res.locals.questionGHLink = `https://github.com/PrairieLearn/PrairieLearn/tree/master/exampleCourse/questions/${res.locals.question.qid}`;
+      questionGHLink = `https://github.com/PrairieLearn/PrairieLearn/tree/master/exampleCourse/questions/${res.locals.question.qid}`;
     }
 
-    res.locals.qids = await sqldb.queryRows(
+    const qids = await sqldb.queryRows(
       sql.qids,
       { course_id: res.locals.course.id },
       z.string().nullable(),
     );
 
-    res.locals.a_with_q_for_all_ci = await sqldb.queryRows(
+    const a_with_q_for_all_ci = await sqldb.queryRows(
       sql.select_assessments_with_question_for_display,
       { question_id: res.locals.question.id },
       SelectedAssessmentsSchema,
     );
-    res.locals.sharing_enabled = await features.enabledFromLocals('question-sharing', res.locals);
+    const sharingEnabled = await features.enabledFromLocals('question-sharing', res.locals);
 
-    if (res.locals.sharing_enabled) {
-      let result = await sqldb.queryAsync(sql.select_sharing_sets, {
+    let sharingSetsIn, sharingSetsOther;
+    if (sharingEnabled) {
+      const result = await sqldb.queryAsync(sql.select_sharing_sets, {
```

Comment:
Consider using Zod validation here.

Reply:
Updated to use `queryRows` instead of `queryAsync` and validating with `SharingSetSchema` in [e8cc26c](https://github.com/PrairieLearn/PrairieLearn/pull/9571/commits/e8cc26c79af51815e4e9bdbf5d82f9f432f5decc).